### ISSUE TITLE
Support of MAVLink 2.0 on SiK Multipoint

### ISSUE THREAD

**Issue by mzahana** [RFDesign/SiK_Multipoint#2]
Hi.

I would like to use the multipoint firmware with MAVLink 2.0, but I am not sure if it supports it. The firmware [here](https://github.com/LorenzMeier/SiK) has some changes to normal SiK firmware (not multipoint). I guess [these commits](https://github.com/LorenzMeier/SiK/commit/bd43a9fe6f5c8b5f90e84ed9ec7114db6aaa0a11) reflect the changes. Can the SiK_Multipoint firmware be updated to support detection of MAVLink 2.0?

Thanks.

**Comment by lhovo** [comment]
If you're able to get the changes working on the multipoint code then send us a pull request and I will add them in.

PS please use the new multipoint code, this report is old.

See readme for more information

**Comment by mzahana** [comment]
Yes I can test now on 3DR 433/915MHz telemetry modules, and later on RFD900+ when they arrive. Thanks.

**Comment by lhovo** [comment]
Ok, try these out..
https://github.com/lhovo/file_share/tree/master/SiK_multipoint_Mav2

I believe the RFD900+ already supports Mavlink2

**Comment by mzahana** [comment]
Thank you very much. I will get a chance to try them out tomorrow (or after) and give you feedback immediately.

**Comment by mzahana** [comment]
@lhovo I tried the ```radio~hm_trp.ihx``` but it does not work!

Here is how I uploaded the file to the 915MHz module
```bash
./uploader.py --port /dev/ttyUSB0 ~/Downloads/file_share-master/SiK_multipoint_Mav2/radio~hm_trp.ihx 
uploading to port /dev/ttyUSB0
Connecting to /dev/ttyUSB0
Trying autosync

+++[0] OK
ATI


ATI
[0] MP SiK 2.6 on
AT&UPDATE
Sent update command
Got sync
board 4e  freq 91
erasing...
programing...
verifying...
done.
```
The LED keep in red state. It does not go to green.

Previously, I used to flash it the same way and it works fine with the following commands

```bash
./uploader.py --port /dev/ttyUSB0 ../dst/radio~hm_trp.ihx 
uploading to port /dev/ttyUSB0
Connecting to /dev/ttyUSB0
Got sync
board 4e  freq 91
erasing...
programing...
[##################################################] 30232/30232 (100%)

verifying...
[##################################################] 30232/30232 (100%)

done.
```

So, what could be the problem?

Thanks.

**Comment by mzahana** [comment]
@lhovo Any idea on the previous issue?

**Comment by lhovo** [comment]
Not sure sorry, I haven't been near my computer to have a look.

Are you able to compile the code?
If so I will upload the changes and you can have a go compiling it.

**Comment by mzahana** [comment]
@lhovo I just used the binaries you provided in the link. Is there any code you modified so I can try to compile?

**Comment by mzahana** [comment]
@lhovo I could compile the old code and flash it. It works fine. But, I don't see a modified code.

**Comment by mzahana** [comment]
I could compile the branch (`make install`) and flash it on both 3DR 915Mhz and RFD900p. No issues so far. I will make some more test in flight soon. Thanks.
